File: frontend/src/lib/types/grading.ts
export interface GradingRequest {
    student_id: string;
}

export interface GradingResponse {
    assignment_id: string;
    student_id: string;
    score: number;
    feedback: string;
    strengths: string[];
    areas_for_improvement: string[];
    missed_concepts: string[];
}

export interface BatchGradingRequest {
    assignment_id: string;
}

export interface BatchGradingResponse {
    batch_id: string;
    results: GradingResponse[];
    completed: boolean;
}

File: backend/app/domain/models/grading.py
from pydantic import BaseModel
from typing import List, Optional

class GradingRequest(BaseModel):
    student_id: str

class GradingResponse(BaseModel):
    assignment_id: str
    student_id: str
    score: int
    feedback: str
    strengths: List[str] = []
    areas_for_improvement: List[str] = []
    missed_concepts: List[str] = []

class GradingSimilarity(BaseModel):
    teacher_text: str
    student_text: str
    similarity: float
    
class BatchGradingRequest(BaseModel):
    student_id: List[str]
    
class BatchGradingResponse(BaseModel):
    batch_id: str
    assignment_id: str
    completed: bool
    results: List[GradingResponse] = []